shell/platform/android/android_context_gl.cc

```diff
@@ -196,13 +155,39 @@ AndroidContextGL::~AndroidContextGL() {
     LogLastEGLError();
   }
 
-  if (!TeardownSurface(environment_->Display(), surface_)) {
-    FML_LOG(ERROR)
-        << "Could not tear down the EGL surface. Possible resource leak.";
+  if (!TeardownContext(environment_->Display(), resource_context_)) {
+    FML_LOG(ERROR) << "Could not tear down the EGL resource context. Possible "
+                      "resource leak.";
     LogLastEGLError();
   }
 }
 
+// For onscreen rendering.
+EGLSurface AndroidContextGL::CreateOnscreenSurface(
```

Comment:
Prefer RAII objects. It is not clear from this method or its documentation about how and when the EGLSurface must be collected. See fml/unique_object.h for one possible utility to implement a unique EGLSurface that cleans up after itself.